File: sorting-searching/sorting-algorithms/bubble-sort/bubble.js
function bubbleSort(arr) {
  const n = arr.length;

  for (let i = 0; i < n; i++) {
    let swapped = false;

    for (let j = 0; j < n - i - 1; j++) {
      if (arr[j] > arr[j + 1]) {
        [arr[j], arr[j + 1]] = [arr[j + 1], arr[j]];
        swapped = true;
      }
    }

    if (!swapped) {
      break;
    }
  }

  return arr;
}

const arr = [5, 3, 8, 4, 2];
console.log("Original:", arr);
bubbleSort(arr);
console.log("Sorted:", arr);

File: sorting-searching/sorting-algorithms/bubble-sort/bubble.py

## Starter code for `bubble.py`
def bubble_sort(arr):
    n = len(arr)

    for i in range(n):
        swapped = False

        for j in range(0, n - i - 1):
            if arr[j] > arr[j + 1]:
                arr[j], arr[j + 1] = arr[j + 1], arr[j]
                swapped = True

        if not swapped:
            break

    return arr


if __name__ == "__main__":
    arr = [5, 3, 8, 4, 2]
    print("Original:", arr)
    bubble_sort(arr)
    print("Sorted:", arr)